src/calliope/postprocess/orchestrator.py

```diff
@@ -0,0 +1,175 @@
+            ]
+            items.sort(key=lambda t: (t[0].order, str(t[0].name)))
+        return items
+
+    def set_enabled(self, name: str, enabled: bool) -> None:
+        """Enable or disable a specific function."""
+        with self._lock:
+            spec, fn = self._entries[name]
+            self._entries[name] = (spec.model_copy(update={"enabled": enabled}), fn)
+
+
+REGISTRY: Final = PostprocessRegistry()
+
+
+def postprocessor(
+    *,
+    name: str | None = None,
+    base_math: Iterable[str],
+    order: int,
+    enabled: bool,
+    overwrite: bool = False,
+) -> Callable[[F], F]:
+    """Decorator used to add new functions to the postprocessing registry.
+
+    Args:
+        base_math (Iterable[str]): compatible base math files for this function.
+        name (str, optional): name to use for this post processing function. Defaults to the function name.
+        order (int): priority order.
+        enabled (bool): whether or not this function should be enabled.
+        overwrite (bool, optional): whether or not to allow overwriting functions. Defaults to False.
+    """
+
+    def decorator(func: F) -> F:
+        REGISTRY.register_function(
+            func,
+            base_math=base_math,
+            name=name or func.__name__,
+            order=order,
+            enabled=enabled,
+            overwrite=overwrite,
+        )
+        return func
+
+    return decorator
+
+
+def _apply_zero_threshold(results: xr.Dataset, zero_threshold: float) -> None:
+    """Remove unreasonably small values in-place.
+
+    Used to avoid floating point errors caused by solver output.
+    Reasonable value = 1e-12.
+    """
+    for name in list(results.data_vars):
+        # If there are any values in the data variable which fall below the
+        # threshold, note the data variable name and set those values to zero
+        results[name] = xr.where(
+            np.abs(results[name]) < zero_threshold, 0, results[name]
+        )
+
+    LOGGER.info(
+        "Postprocessing: applied zero threshold %s to model results.", zero_threshold
+    )
+
+
+def postprocess_results(
+    results: xr.Dataset, inputs: xr.Dataset, config: CalliopeConfig
+) -> xr.Dataset:
+    """Run compatible postprocessors in deterministic order."""
+    if config.solve.enable_postprocessing:
+        applicable_postprocesses = REGISTRY.get_applicable(config.init.base_math)
+        ctx = PostprocessContext(config=config, inputs=inputs, results=results)
+
+        for settings, function in applicable_postprocesses:
+            name = settings.name
+            if (name in results) or name in (results.dims):
+                raise ValueError(
+                    f"Postprocess '{name}' attempted to overwrite an existing variable/coord/dim."
+                )
+            try:
+                # Store postprocesses in the model results.
+                LOGGER.info("Postprocessing: applying %s postprocess function.", name)
```

Comment:
Rogue to not choose an f-string!
```suggestion
                LOGGER.info(f"Postprocessing: applying {name} postprocess function.")
```

Reply:
Not rogue at all ;)
https://google.github.io/styleguide/pyguide.html#3101-logging